src/scripts/main.js

```diff
@@ -1 +1,63 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let hasLeftClick = false;
+  let hasRightClick = false;
+
+  document.addEventListener('click', () => {
+    hasLeftClick = true;
+
+    if (hasLeftClick && hasRightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', () => {
+    hasRightClick = true;
+
+    if (hasLeftClick && hasRightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
```

Comment:
The third promise resolves when both a left-click and a right-click have occurred. However, it will resolve twice if both events occur, as the resolve function is called in both event listeners. Consider using a flag to ensure it resolves only once.